src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/Writer.java

```diff
@@ -51,7 +51,7 @@ public Writer(Path dest, List<? extends JavaFileObject> files) {
         } else {
             return InMemoryJavaCompiler.compile(sources,
                 "--add-modules", "jdk.incubator.foreign",
-                "--release", "16",
+                "--release", "17",
```

Comment:
Note that this is required to see the new linker method to create virtual handles - but this creates a dependency with 17 - e.g. the generated API will no longer work against 16.